dilithium.py

```diff
@@ -427,7 +432,10 @@ def sign(self, sk_bytes, m):
         # Set seeds and nonce (kappa)
         mu = self._h(tr + m, 64)
         kappa = 0
-        rho_prime = self._h(K + mu, 64)
+        if rnd is None:
+            # deterministic signing
+            rnd = b"\x00"*32  # RNDBYTES
```

Comment:
```suggestion
            rnd = self.random_bytes(32)
```
Currently, this change would result in the "deterministic variant" to be "the default". However, from my point of view and understanding of the Draft-FIPS-204, according to line 682f, the hedged version should be the default with fresh randomness.

Reply:
I understand where you're coming from, I was just matching the default from the C reference implementation's standard branch. There, you need to enable randomized signing by setting `DILITHIUM_RANDOMIZED_SIGNING`. 
This change would keep the self-test functional but break the KATs, right?